.github/workflows/test.yml

```diff
@@ -1,38 +1,49 @@
-on: push
+on:
+  push:
+  pull_request:
+    types: [synchronize]
 name: Test
 jobs:
   test:
     strategy:
       fail-fast: false
       matrix:
-        go-version: [1.16.x, 1.17.x]
+        go-version: [1.17.x, 1.18.x]
         platform: [ubuntu-latest, macos-latest, windows-latest]
     runs-on: ${{ matrix.platform }}
     steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
       - name: Install Go
-        uses: actions/setup-go@v1
+        uses: actions/setup-go@v2
         with:
           go-version: ${{ matrix.go-version }}
-      - name: Install staticcheck
-        run: go install honnef.co/go/tools/cmd/staticcheck@latest
-        shell: bash
-      - name: Install golint
-        run: go install golang.org/x/lint/golint@latest
-        shell: bash
-      - name: Update PATH
-        run: echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
-        shell: bash
-      - name: Checkout code
-        uses: actions/checkout@v1
       - name: Fmt
         if: matrix.platform != 'windows-latest' # :(
         run: "diff <(gofmt -d .) <(printf '')"
         shell: bash
       - name: Vet
         run: go vet ./...
       - name: Staticcheck
-        run: staticcheck ./...
-      - name: Lint
-        run: golint ./...
```

Comment:
Lint seems to just go away. Is that what we want?

Reply:
According to the [golint](https://github.com/golang/lint) readme:

>NOTE: Golint is [deprecated and frozen](https://github.com/golang/go/issues/38968). There's no drop-in replacement for it, but tools such as [Staticcheck](https://staticcheck.io/) and go vet should be used instead.

So, yes, IMO we should stop using it. But Staticcheck's style checks are not usually enabled by default, so we might need to change something if we want its more lint-like features. Something to pass the equivalent of `-checks=all` perhaps. That can also be accomplished by a [staticcheck.conf file](https://staticcheck.io/docs/configuration/#configuration-files) to configure it for anyone running `staticcheck` on a clone of the repo as well.

Reply:
Sure, understood. I'll propose `gofumpt` in a separate PR.

Reply:
Go for it, I'm a fan of `gofumpt`.